gpt4all-chat/tests/python/test_server_api.py

```diff
@@ -23,50 +25,81 @@ def __init__(self) -> None:
     def get(self, path: str, *, wait: bool = False) -> Any:
         return self._request('GET', path, wait)
 
-    def _request(self, method: str, path: str, wait: bool) -> Any:
+    def post(self, path: str, data: dict[str, Any] | None, *, wait: bool = False) -> Any:
+        return self._request('POST', path, wait, data)
+
+    def _request(self, method: str, path: str, wait: bool, data: dict[str, Any] | None = None) -> Any:
         if wait:
             retry = Retry(total=None, connect=10, read=False, status=0, other=0, backoff_factor=.01)
         else:
             retry = Retry(total=False)
         self.http_adapter.max_retries = retry  # type: ignore[attr-defined]
 
-        resp = self.session.request(method, f'http://localhost:4891/v1/{path}')
+        resp = self.session.request(method, f'http://localhost:4891/v1/{path}', json=data)
         resp.raise_for_status()
         return resp.json()
 
 
 request = Requestor()
 
 
+def create_chat_server_config(tmpdir: Path, model_copied: bool = False) -> dict[str, str]:
+    xdg_confdir = tmpdir / 'config'
+    app_confdir = xdg_confdir / 'nomic.ai'
+    app_confdir.mkdir(parents=True)
+    with open(app_confdir / 'GPT4All.ini', 'w') as conf:
+        conf.write(textwrap.dedent(f"""\
+            [General]
+            serverChat=true
+
+            [download]
+            lastVersionStarted={config.APP_VERSION}
+
+            [network]
+            isActive=false
+            usageStatsActive=false
+        """))
+
+    if model_copied:
+        app_data_dir = tmpdir / 'share' / 'nomic.ai' / 'GPT4All'
+        app_data_dir.mkdir(parents=True)
+        local_file_path_env = os.getenv('TEST_MODEL_PATH')
+        if local_file_path_env:
+            local_file_path = Path(local_file_path_env)
+            if local_file_path.exists():
+                shutil.copy(local_file_path, app_data_dir / local_file_path.name)
+            else:
+                pytest.fail(f'Model file specified in TEST_MODEL_PATH does not exist: {local_file_path}')
+        else:
+            pytest.fail('Environment variable TEST_MODEL_PATH is not set')
```

Comment:
It's better to ask for forgiveness than for permission (in Python, at least). This could be simply:
```suggestion
        shutil.copy(os.environ['TEST_MODEL_PATH'], app_data_dir / local_file_path.name)
```
And an informative KeyError or FileNotFoundError will be raised if there is a problem. This also clarifies that this is not something being tested (pass/fail), but an expectation about the environment the test itself runs in.

Reply:
+1, except that local_file_path isn't quite right.
```py
local_env_file_path = Path(os.environ['TEST_MODEL_PATH'])
shutil.copy(local_env_file_path, app_data_dir / local_env_file_path.name)
```